### PR TITLE
Add infrastructure permissions docs

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**How to categorize this PR?**

/area documentation
/kind enhancement
/priority normal
/platform gcp

**What this PR does / why we need it**:

Add infrastructure permissions docs.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: docs/usage-as-end-user.md [hunk 1]
```diff
@@ -4,9 +4,27 @@ The [`core.gardener.cloud/v1beta1.Shoot` resource](https://github.com/gardener/g
...
 In this document we are describing how this configuration looks like for GCP and provide an example `Shoot` manifest with minimal configuration that you can use to create an GCP cluster (modulo the landscape-specific information like cloud profile names, secret binding names, etc.).
 
-## Provider Secret Data
+## GCP Provider Credentials
+
+In order for Gardener to create a Kubernetes cluster using GCP infrastructure components, a Shoot has to provide credentials with sufficient permissions to the desired GCP project.
+Every shoot cluster references a `SecretBinding` which itself references a `Secret`, and this `Secret` contains the provider credentials of the GCP project.
+The `SecretBinding` is configurable in the [Shoot cluster](https://github.com/gardener/gardener/blob/master/example/90-shoot.yaml)) with the field `secretBindingName`.
+
+The required credentials for the GCP project are a [Service Account Key](https://cloud.google.com/iam/docs/service-accounts#service_account_keys)) to authenticate as a [GCP Service Account](https://cloud.google.com/compute/docs/access/service-accounts).
```
Comment (DockToFuture) [inline docs/usage-as-end-user.md:13]: just a small typo, remove one ")"

**Review by DockToFuture — CHANGES_REQUESTED** [review]

File: docs/usage-as-end-user.md [hunk 2]
```diff
@@ -4,9 +4,27 @@ The [`core.gardener.cloud/v1beta1.Shoot` resource](https://github.com/gardener/g
 
 In this document we are describing how this configuration looks like for GCP and provide an example `Shoot` manifest with minimal configuration that you can use to create an GCP cluster (modulo the landscape-specific information like cloud profile names, secret binding names, etc.).
 
-## Provider Secret Data
+## GCP Provider Credentials
+
+In order for Gardener to create a Kubernetes cluster using GCP infrastructure components, a Shoot has to provide credentials with sufficient permissions to the desired GCP project.
+Every shoot cluster references a `SecretBinding` which itself references a `Secret`, and this `Secret` contains the provider credentials of the GCP project.
+The `SecretBinding` is configurable in the [Shoot cluster](https://github.com/gardener/gardener/blob/master/example/90-shoot.yaml)) with the field `secretBindingName`.
```
Comment (DockToFuture) [inline docs/usage-as-end-user.md:11]: just a small typo, remove one ")"

**Review by DockToFuture — CHANGES_REQUESTED** [review]

**Review by rfranzke — COMMENTED** [review]
Looks great after addressing the nit mentioned by @DockToFuture

File: docs/usage-as-operator.md [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+Similarly, the [`core.gardener.cloud/v1beta1.Seed` resource](https://github.com/gardener/gardener/blob/master/example/50-seed.yaml) is structured.
+Additionally, it allows to configure settings for the backups of the main etcds' data of shoot clusters control planes running in this seed cluster.
 
+This document explains the necessary configuration for this provider extension.
 In this document we are describing how this configuration looks like for GCP and provide an example `CloudProfile` manifest with minimal configuration that you can use to allow creating GCP shoot clusters.
 
-## `CloudProfileConfig`
+## `CloudProfile` resource
+
+In this section we are describing how the configuration for `CloudProfile`s looks like for GCP and provide an example `CloudProfile` manifest with minimal configuration that you can use to allow creating GCP shoot clusters.
```
Comment (prashanth26) [inline docs/usage-as-operator.md:12]:
```suggestion
In this section, we are describing how the configuration for `CloudProfile`s looks like for GCP and provide an example `CloudProfile` manifest with a minimal configuration that you can use to allow creating GCP shoot clusters.
```

**Review by prashanth26 — COMMENTED** [review]
Thanks for the PR. Looks great. Minor suggestions.

**Review by prashanth26 — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]

File: docs/usage-as-end-user.md [hunk 4]
```diff
@@ -2,11 +2,29 @@
...
+This document describes the configurable options for GCP and provides an example `Shoot` manifest with minimal configuration that can be used to create a GCP cluster (modulo the landscape-specific information like cloud profile names, secret binding names, etc.).
 
-## Provider Secret Data
+## GCP Provider Credentials
+
+In order for Gardener to create a Kubernetes cluster using GCP infrastructure components, a Shoot has to provide credentials with sufficient permissions to the desired GCP project.
+Every shoot cluster references a `SecretBinding` which itself references a `Secret`, and this `Secret` contains the provider credentials of the GCP project.
+The `SecretBinding` is configurable in the [Shoot cluster](https://github.com/gardener/gardener/blob/master/example/90-shoot.yaml) with the field `secretBindingName`.
+
+The required credentials for the GCP project are a [Service Account Key](https://cloud.google.com/iam/docs/service-accounts#service_account_keys)) to authenticate as a [GCP Service Account](https://cloud.google.com/compute/docs/access/service-accounts).
```
Comment (DockToFuture) [inline docs/usage-as-end-user.md:13]: Can you please remove one ")"?

**Review by rfranzke — APPROVED** [review]

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Pull request merged**